### PR TITLE
STM32F0xxGxUx and QFN-28-ST

### PR DESCRIPTION
**Pull request by HarkonenBade**
Adds a library part for the STM32F0xxGxUx and an associated footprint for the QFN-28-ST. Footprint was initially generated using build-mod-ic, but was then hand edited to account for the chamfered corner pads required by the data sheet. These are currently implemented as a set of identically numbered pads as shown in the third image below. Kicad seems to accept this and the footprint seems to operate correctly. It does not violate DRC with a 0.2mm clearance as standard. It does identify the extra pads as unconnected sadly. I'm looking to see if anything can be done about this, but I suspect not until Kicad supports polygon pads.

[Datasheet](http://www.st.com/content/ccc/resource/technical/document/datasheet/52/ad/d0/80/e6/be/40/ad/DM00105814.pdf/files/DM00105814.pdf/jcr:content/translations/en.DM00105814.pdf)

screen shot 2017-08-22 at 22 47 37
screen shot 2017-08-22 at 22 40 41
screen shot 2017-08-22 at 22 40 52
screen shot 2017-08-22 at 22 36 10
screen shot 2017-08-22 at 22 41 19

### CODE REVIEW

**Comment by HarkonenBade** [comment]
Have worked out a new arraignment of overlapping pads which does not produce the same unconnected warnings, will update the PR later today with a new footprint.

File: lib/ic/microcontroller/stm32f0xxgxux.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+name: STM32F0xxGxUx
+ordercodes:
+- [Farnell, '2469551']
+pins:
+- - - [VDD, 17, pwrin]
+    - [VDDIO2, 18, pwrin]
+    - [VDDA, 5, pwrin]
+    - [VSS, 16, pwrin]
+  - - [NRST, 4, in]
+    - [BOOT0, 1, in]
```
Comment (adamgreig) [inline lib/ic/microcontroller/stm32f0xxgxux.yaml]: Please include that this is also PB8, eg `PB8-BOOT0`

File: lib/ic/microcontroller/stm32f0xxgxux.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- - - [PA0, 6, bidi]
+    - [PA1, 7, bidi]
+    - [PA2, 8, bidi]
+    - [PA3, 9, bidi]
+    - [PA4, 10, bidi]
+    - [PA5, 11, bidi]
+    - [PA6, 12, bidi]
+    - [PA7, 13, bidi]
+    - [PA9, 19, bidi]
+    - [PA10, 20, bidi]
```
Comment (adamgreig) [inline lib/ic/microcontroller/stm32f0xxgxux.yaml]: Please indicate that PA9 and PA10 can also be PA11 and PA12, probably `PA9-PA11` and `PA10-PA12`.

File: lib/ic/microcontroller/stm32f0xxgxux.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+- [Farnell, '2469551']
+pins:
+- - - [VDD, 17, pwrin]
+    - [VDDIO2, 18, pwrin]
+    - [VDDA, 5, pwrin]
+    - [VSS, 16, pwrin]
+  - - [NRST, 4, in]
+    - [BOOT0, 1, in]
+  - - [PF0, 2, bidi]
+    - [PF1, 3, bidi]
```
Comment (adamgreig) [inline lib/ic/microcontroller/stm32f0xxgxux.yaml]: Please include that these are OSC_IN and OSC_OUT: `PF0-OSC_IN` and `PF1-OSC_OUT`.

**Comment by HarkonenBade** [comment]
Done

**Comment by adamgreig** [comment]
Thanks!

**Pull request merged**